### PR TITLE
Read me

### PR DESCRIPTION
**Pull request by aepaysinger**
it has been so long since i did this i wanted to make sure this was looking okay

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,39 @@
-# code-challenges-javascript
\ No newline at end of file
+# code-challenges-javascript
+This repo is for Javascript code-challenges
+## Set up
+1) cd into the directory you want this repository in - clone it down
+```
+git clone <path>
```
Comment (W-Ely) [inline README.md:6]:
```suggestion
glt clone https://github.com/aepaysinger/code-challenges-javascript.git
```

File: README.md [hunk 2]
```diff
@@ -1 +1,39 @@
...
+```
+2) install jest for testing
+```
```
Comment (W-Ely) [inline README.md:9]: This is incorrect.  `jest` is already listed as a development dependency in your [package.json ](https://github.com/aepaysinger/code-challenges-javascript/blob/main/package.json#L20)

JavaScript how to install dev dependencies package.json.

File: README.md [hunk 3]
```diff
@@ -1 +1,39 @@
...
+npm install --save-dev jest
+```
+3) in package.json
+ ``` {
+  "scripts": {
+    "test": "jest"
+  }
+}
```
Comment (W-Ely) [inline README.md:17]: What is this step?  If they cloned down the repo, the package.json is the one from the repo.

**Review by W-Ely — CHANGES_REQUESTED** [review]
Minor changes

**Comment by aepaysinger** [comment]
is there anything else i need to change?

**Comment by W-Ely** [comment]
Steps 2 and 3 are still wrong.

**Comment by aepaysinger** [comment]
so i updated it a while ago and didnt realize i was having issues with pushing it through. when i go to git push read-me, it says:  ! [rejected]        read-me -> read-me (non-fast-forward)
error: failed to push some refs to 'https://github.com/aepaysinger/code-challenges-javascript.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
i have tried merging with main but it is already up to date.....

**Comment by W-Ely** [comment]
> i have tried merging with main but it is already up to date.....

What does the `hint` mean to you?  Have you checked Github to see what the read-me branch looks like?

**Comment by aepaysinger** [comment]
i have checked git hub and the read me branch doesnt look the way i want it too. i need to push what i have on the branch locally. 
im not sure what it means by the tip of current branch is behind.
do i need to checkout out to main and then git pull read-me?

**Comment by W-Ely** [comment]
Are the few file deletions intended here?

**Comment by aepaysinger** [comment]
yes.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,34 @@
...
-### Convert boolean To String:
+This repo is for Javascript code-challenges
+## Set up
+1) cd into the directory you want this repository in - clone it down
+```
+git clone https://github.com/aepaysinger/code-challenges-javascript.git
+```
+2) install devDependencies for testing
+```
+npm install --production
```
Comment (W-Ely) [inline README.md:10]: What does `--production` do?  Didn't work for me
```
[6:27:59] ➜  code-challenges-javascript git:(main) npm install --production
npm WARN config production Use `--omit=dev` instead.

up to date, audited 1 package in 362ms

found 0 vulnerabilities
[6:28:07] ➜  code-challenges-javascript git:(main) npm test

> code-challenges-javascript@1.0.0 test
> jest

sh: jest: command not found
[6:28:28] ➜  code-challenges-javascript git:(main)
```
https://docs.npmjs.com/cli/v9/commands/npm-install

**Review by W-Ely — COMMENTED** [review]
> yes.

Why?

[src/challenges/price_of_magoes.js](https://github.com/aepaysinger/code-challenges-javascript/pull/17/files#diff-d53443bba1f4066f2087963947ea99cbd87ed7e1e33e422a389a31e70ddb49a7) -- This file has a corresponding .test.js file.  If you remove it main will be in a failing state.

[src/tests/example.test.js](https://github.com/aepaysinger/code-challenges-javascript/pull/17/files#diff-340bf2619b23e272115ba5645b54d75b0c643283d208705d4bfd8bd50ffb4a10) -- example.js remains so why delete the test?

[src/tests/reduce_but_grow.test.js](https://github.com/aepaysinger/code-challenges-javascript/pull/17/files#diff-257d1e983b9bd0e762451eabdb208439556730b8219b727d78934f2c4d3cb049) -- same reduce_but_grow remains so why delete the test?

> it has been so long since i did this i wanted to make sure this was looking okay
VScode lets you preview markdown locally, have you taken a look?  also you can navigate to the branch here https://github.com/aepaysinger/code-challenges-javascript/tree/read-me to see how it will actually look in Github.

When writing instructions for setup, you should make an effort to ensure that they actually work by at least cloning down the repo fresh or into a new location and walking the steps you lay out.

File: README.md [hunk 5]
```diff
@@ -1,10 +1,23 @@
 # code-challenges-javascript
-This repo is for Javascript code-challenges
+This repo is for Javascript code-challenges.
+You will need to have node installed locally
 ## Set up
-cd into the directory you want this repository in - clone it down
+move into the directory you want this repository in 
+```
+cd name of directory
```
Comment (W-Ely) [inline README.md:7]:
```suggestion
cd <path to directory>
```

**Review by W-Ely — APPROVED** [review]

**Comment by W-Ely** [comment]
> Are the few file deletions intended here?
> >yes.

> Why?
[src/challenges/price_of_magoes.js](https://github.com/aepaysinger/code-challenges-javascript/pull/17/files#diff-d53443bba1f4066f2087963947ea99cbd87ed7e1e33e422a389a31e70ddb49a7) -- This file has a corresponding .test.js file. If you remove it main will be in a failing state.
[src/tests/example.test.js](https://github.com/aepaysinger/code-challenges-javascript/pull/17/files#diff-340bf2619b23e272115ba5645b54d75b0c643283d208705d4bfd8bd50ffb4a10) -- example.js remains so why delete the test?
[src/tests/reduce_but_grow.test.js](https://github.com/aepaysinger/code-challenges-javascript/pull/17/files#diff-257d1e983b9bd0e762451eabdb208439556730b8219b727d78934f2c4d3cb049) -- same reduce_but_grow remains so why delete the test?

Still applies, but the readme now looks good. This is also now an additional unrelated file addition.

**Comment by aepaysinger** [comment]
if its already on main would deleting it from this branch delete from main?

**Pull request closed without merging**